plugin-scalebar/src/main/java/com/mapbox/pluginscalebar/ScaleBarWidget.java

```diff
@@ -0,0 +1,215 @@
+package com.mapbox.pluginscalebar;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.os.Handler;
+import android.os.Message;
+import android.support.annotation.ColorRes;
+import android.support.annotation.VisibleForTesting;
+import android.util.Pair;
+import android.view.View;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Locale;
+
+/**
+ * The scale widget is a visual representation of the scale bar plugin.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+public class ScaleBarWidget extends View {
+  private static int MSG_WHAT = 0;
+  private static int REFRESH_DURATION = 300;
+  private int textColor;
+  private int primaryColor;
+  private int secondaryColor;
+  private int mapViewWidth;
+  private int margin;
+  private int maxBarWidth;
+  private double distancePerPixel;
+  private final Paint textPaint = new Paint();
+  private final Paint barPaint = new Paint();
+  private boolean isMetricSystem;
+  private ArrayList<Pair<Integer, Integer>> scaleTable;
+  private String unit;
+  private RefreshHandler refreshHandler;
+  private Context context;
+
+  public ScaleBarWidget(Context context) {
+    super(context);
+    this.context = context;
+    textPaint.setAntiAlias(true);
+    textPaint.setTextSize(20);
+    textPaint.setTextAlign(Paint.Align.CENTER);
+    barPaint.setAntiAlias(true);
+    textColor = context.getResources().getColor(android.R.color.black);
+    primaryColor = context.getResources().getColor(android.R.color.black);
+    secondaryColor = context.getResources().getColor(android.R.color.white);
+    isMetricSystem = LocaleUnitResolver.isMetricSystem();
```

Comment:
Could you expose a setter for `isMetricSystem`, besides loading a default value from `Locale`?